### DISCUSSION TITLE
User-friendly domain name ownership

### DISCUSSION THREAD

**Discussion by erlend-sh** [muni-town/weird#24] [Ideas]
We will approach this via a partnership and tight integration with [takingnames.io](https://takingnames.io) —
https://takingnames.io/blog/introducing-takingnames-io

That will allow us to do [something like this](https://bsky.social/about/blog/7-05-2023-namecheap), except with eve greater user agency.

@anderspitman can elaborate.

### Free Domains

This work extends to free domains. 🆓 is a major accessibility feature, especially essential for the [unbanked](https://www.npr.org/2021/04/05/984475870/unbanked-what-it-means-to-be-outside-of-the-banking-system).

Here's a cool MVP take on it: https://home.obl.ong

**Comment by anderspitman** [comment]
The key thing to figure out here is the balance between the Weird UX, and the overall UX of users managing domains. Here's what I mean.

The ideal Weird UX would be something very similar to what Bluesky made: a tightly integrated dashboard for buying a domain that is automatically set up as your Bluesky handle. The problem is this only works for one app: Bluesky (or Weird if we build something similar). What if you have a domain and you want to use one subdomain for Bluesky, one for Weird, one for Mastodon provided by masto.host, etc. Oh and you also want to give Fastmail control over the MX records for your apex domain for setting up email, but don't want them to have any other abilities.

Fortunately we've already solve this sort of delegation dilemma with OAuth2. So I should be able to go to Bluesky, and they show me a button that says "Connect a domain from Namecheap" (or Porkbun, or Ghandi, etc), and it does an OAuth2 flow to namecheap.com where it asks me if I want to give Bluesky control over bsky.apitman.com. I say yes, and Bluesky receives a token that lets them call an API to make the necessary DNS record changes. The user doesn't even need to know what a DNS record is. You could implement an entire domain registrar that doesn't even provide the ability to manually edit records.

The problem is that no such DNS OAuth2 protocol is in widespread use today. [DomainConnect](https://www.domainconnect.org/) made an attempt, but it ended up really only being useful for big players, and basically doesn't work for self-hosted services.

This is essentially why I created TakingNames.io, as a place to implement the NameDrop protocol to accomplish this. It works great (you can see a demo on the link @erlend-sh posted), but no one uses it. I never ended up making a big marketing push to get adoption because I'm not particularly interested in being a domain reseller and I've been focusing on other important parts of the self-hosting stack. However, I'm back to thinking this is an important problem that needs to be solved, and Weird could be the perfect place to pilot it.

So basically if we're ok with redirecting to TakingNames.io for granting access to domains, I think we're good to go. I just need to do some work to make TN.io more reliable and probably improve the NameDrop API based on your feedback. I'm very open to doing a revenue share in order to make the finances work.

Alternatively if you want something more integrated I'm open to discussing, but I'm not sure yet how it could be done while solving the many apps problem.

**Reply by erlend-sh** [reply]
> So basically if we're ok with redirecting to TakingNames.io for granting access to domains, I think we're good to go.

Good to go then!
